substrate/frame/system/src/lib.rs

```diff
@@ -401,6 +415,9 @@ pub mod pallet {
 
 		/// The maximum number of consumers allowed on a single account.
 		type MaxConsumers: ConsumerLimits;
+
+		/// State verison used to derive extrinsics root.
+		type ExtrinsicsRootStateVersion: Get<sp_core::storage::StateVersion>;
```

Comment:
No idea yet if doable or fine, but would it be suitable to put this state version as part of RuntimeVersion (type Version) as a new field or a specific api version (as a specific api version, could default to 0 when missing so there would be no need to change existing pallets).